mp/importer/batcher.py

```diff
@@ -0,0 +1,51 @@
+# this function is intentionally get empty for now
+# this program takes an iterable and splits in a number of batches using the batch_size argument (from context)
+# result is a list that contains the batches (for now)
+
+def run_in_batches(context, iterable, process_one): 		
+    num = int(len(iterable))                                      
+    batch_size = context.batch_size
+    start_batch = context.start_batch					
+    batch_stop, extra = divmod(num, batch_size)
+    if start_batch == 0:
+        list = []
+        if extra == 0:								
+            for n in range(0,batch_stop):												
+                batch_item = iterable[n*batch_size:batch_size+n*batch_size] 	
+                result = process_one(batch_item)
+                list.append(result)
```

Comment:
if the number of results is large, storing them all in a list is probably going to cause memory problems. Perhaps you might think about returning an iterator so you don't need to store them all in memory?